src/util.rs

```diff
@@ -0,0 +1,47 @@
+use crate::edwards_point::SonnyEdwardsPointGadget;
+use bulletproofs::r1cs::{Prover, Verifier};
+use curve25519_dalek::{ristretto::CompressedRistretto, scalar::Scalar};
+use zerocaf::edwards::EdwardsPoint as SonnyEdwardsPoint;
+
+/// Helper methods that are exposed to the end-user to allow them  
+/// to convert from a zerocaf type into a gadget specified by this library
+pub fn prover_commit_to_sonny_point(
```

Comment:
We don't need to commit the points since they are publicly known. So why changing the method we had before that just converted the coordinates into LinearCombination for a method that commits theese coordinates when we don't need it?

Reply:
You are correct. We do not need to commit to the points, these are just for tests and shows the end user, how to prove "P1 + P2 = P3" where P1 and P2 are private.

I should add an extra method that just does the conversion for public points.

I believe this comment links to your first in that, the user will decide whether his proof involves public/private points as he is constructing the circuit himself.

Reply:
So we do not plan to provide `prove and verify` functions for the end-user right? Then if the objective is just providing the gadgets I agree that this is the way of doing it, showing how to create a `proof` with commited points.